awx/ui_next/src/components/MultiSelect/MultiSelect.jsx

```diff
@@ -84,33 +82,57 @@ class MultiSelect extends Component {
     document.removeEventListener('mousedown', this.handleClick, false);
   }
 
-  getInitialChipItems() {
-    const { associatedItems } = this.props;
-    return associatedItems.map(item => ({ ...item }));
-  }
-
   handleClick(e, option) {
     if (this.node && this.node.contains(e.target)) {
       if (option) {
-        this.handleSelection(e, option);
+        e.preventDefault();
+        this.addItem(option);
       }
     } else {
       this.setState({ input: '', isExpanded: false });
     }
   }
 
```

Comment:
This might be a better way of keeping track of that event listener on the DOM and only applying when necessary.  ```  handleClick(e, option) {
    document.addEventListener('mousedown', this.handleClick, false);
    if (this.node && this.node.contains(e.target)) {
      if (option) {
        e.preventDefault();
        this.addItem(option);
        document.removeEventListener('mousedown', this.handleClick, false);
      }
    } else {
      this.setState({ input: '', isExpanded: false });
      document.removeEventListener('mousedown', this.handleClick, false);
    }
  }```

This will allow you to remove the add event listener from `componentDIdMount`.  You would also need to add `onClick={this.handleClick}` to the `TextInput` component in render.

Reply:
No, you don't want to add a new listener on every click. You only want to add one on mount and remove it on unmount. Otherwise, you end up with multiple listeners and a memory leak.